src/roiextractors/extractors/femtonicsimagingextractor/femtonicsimagingextractor.py

```diff
@@ -0,0 +1,343 @@
+        self._setup_video_data()
+
+    def _setup_channel_selection(self):
+        """Determine which channel to extract using either name or index."""
+        available_channels = self.get_available_channels_from_file()
+
+        if self._channel_name is not None:
+            if self._channel_name not in available_channels:
+                raise ValueError(f"Channel '{self._channel_name}' not found. Available: {available_channels}")
+            self._selected_channel_index = available_channels.index(self._channel_name)
+            self._selected_channel_name = self._channel_name
+        else:
+            if self._channel_index >= len(available_channels):
+                raise ValueError(f"Channel index {self._channel_index} out of range. Available: {available_channels}")
+            self._selected_channel_index = self._channel_index
+            self._selected_channel_name = available_channels[self._channel_index]
+
+    def _setup_video_data(self):
+        """Setup access to the actual imagon data  for the selected measurement unit and chanel."""
+        session_key = f"MSession_{self._munit}"
+        munit_key = f"MUnit_{self._munit}"
+        channel_key = f"Channel_{self._selected_channel_index}"
+
+        if (
+            session_key in self._file
+            and munit_key in self._file[session_key]
+            and channel_key in self._file[session_key][munit_key]
+        ):
+            self._video = DatasetView(self._file[session_key][munit_key][channel_key])
+        else:
+            raise Exception(f"Cannot find data at {session_key}/{munit_key}/{channel_key}")
+
+    def get_available_channels_from_file(self) -> List[str]:
+        """Get available channels from the current file."""
+        session_key = f"MSession_{self._munit}"
+        munit_key = f"MUnit_{self._munit}"
+
+        if session_key not in self._file or munit_key not in self._file[session_key]:
+            raise ValueError(f"MUnit {self._munit} not found in file")
+
+        attrs = dict(self._file[session_key][munit_key].attrs)
+        num_channels = attrs.get("VecChannelsSize", 2)
+
+        channels = []
+        for i in range(num_channels):
+            name_arr = attrs.get(f"Channel_{i}_Name", [])
+            name = self._decode_string(name_arr) if len(name_arr) > 0 else f"Channel_{i}"
+            channels.append(name)
+        return channels
+
+    def _decode_string(self, arr) -> str:
+        """Convert int16 array to string."""
+        if len(arr) == 0:
+            return ""
+        return "".join(chr(x) for x in arr if x != 0)
+
+    def get_image_shape(self) -> Tuple[int, int]:
+        """Get the shape of the video frame (num_rows, num_columns)."""
+        return self._video.shape[1], self._video.shape[2]  # (height, width)
+
+    def get_num_samples(self) -> int:
+        """Get the number of samples (frames) in the video."""
+        return self._video.shape[0]
+
+    def get_channel_names(self) -> List[str]:
```

Comment:
There should be also a static method that allows the user to see which channels are available so they can initialize the extractor. See here for an example:

https://github.com/catalystneuro/roiextractors/blob/afd085ab9d784d3e9c6fb6f2e27f031e803cc6d2/src/roiextractors/extractors/tiffimagingextractors/scanimagetiffimagingextractor.py#L699-L720

Reply:
Oh, you did below. Thanks.